src/lister_haxe/test/lister_test.hx

```diff
@@ -0,0 +1,36 @@
+package test;
+
+import haxe.unit.TestCase;
+using StringTools;
+using test.TestCaseExtender;
+
+class Lister_Test extends TestCase
+{
+    // It ensures returned urls are fully qualified
+    @:access(Lister)
+    public function testUrlsAreFullyQualified() {
+        var host = "http://www.example.com";
+        var actual = Lister.clean("/a.png", host);
+        assertTrue(actual.startsWith(host));
+    }
+
+    // It ensures paths are clean
+    @:access(Lister)
+    public function testUrlsAreClean() {
+        var host = "http://www.example.com/";
+        var actual = Lister.clean("/a.png", host).replace("http://", "");
```

Comment:
Looking at this now, I should just take the protocol specification out of `host`, and drop the `replace()` call.